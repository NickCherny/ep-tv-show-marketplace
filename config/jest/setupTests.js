import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn(),
};

global.localStorage = localStorageMock;

export const __mock_data__ = {
  shows: [{
    id: 250,
    url: "http://www.tvmaze.com/shows/250/kirby-buckets",
    name: "Kirby Buckets",
    type: "Scripted",
    language: "English",
    genres: ["Comedy"],
    status: "Ended",
    runtime: 30,
    premiered: "2014-10-20",
    officialSite: "http://disneyxd.disney.com/kirby-buckets",
    schedule: {time: "07:00", days: Array(5)},
    rating: {average: null},
    weight: 0,
    image: {
      medium: "http://static.tvmaze.com/uploads/images/medium_portrait/1/4600.jpg",
      original: "http://static.tvmaze.com/uploads/images/original_untouched/1/4600.jpg"
    },
    summary: "<p>The single-camera series that mixes live-action and animation stars Jacob Bertrand as the title character.</p>",
    updated: 1574820377,
    episodes: [
      {
        id: "20849",
        url: "http://www.tvmaze.com/episodes/20849/kirby-buckets-1x01-cars-buses-and-lawnmowers",
        name: "Cars, Buses and Lawnmowers",
        season: 1,
        number: 1,
        airdate: "2014-10-20",
        airtime: "20:00",
        airstamp: "2014-10-21T00:00:00+00:00",
        runtime: 30,
        image: null,
        summary: "<p>Kirby learns about a character design contest featuring the </p>",
      },
      {
        id: "20850",
        url: "http://www.tvmaze.com/episodes/20849/kirby-buckets-1x01-cars-buses-and-lawnmowers",
        name: "Cars, Buses and Lawnmowers",
        season: 1,
        number: 1,
        airdate: "2014-10-20",
        airtime: "20:00",
        airstamp: "2014-10-21T00:00:00+00:00",
        runtime: 30,
        image: null,
        summary: "<p>Kirby learns about a character design contest featuring the </p>",
      }
    ]
  },
    {
      id: 251,
      url: "http://www.tvmaze.com/shows/251/downton-abbey",
      name: "Downton Abbey",
      type: "Scripted",
      language: "English",
      genres: (3) ["Drama", "Family", "Romance"],
      status: "Ended",
      runtime: 60,
      premiered: "2010-09-26",
      officialSite: "http://www.itv.com/downtonabbey",
      schedule: {time: "21:00", days: Array(1)},
      rating: {average: 9.1},
      weight: 93,
      webChannel: null,
      externals: {tvrage: 26615, thetvdb: 193131, imdb: "tt1606375"},
      image: {
        medium: "http://static.tvmaze.com/uploads/images/medium_portrait/1/4601.jpg",
        original: "http://static.tvmaze.com/uploads/images/original_untouched/1/4601.jpg"
      },
      summary: "<p>The Downton Abbey estate stands a splendid example of confidence and mettle.</p>",
      updated: 1573501184,
      episodes: [
        {
          id: "20851",
          url: "http://www.tvmaze.com/episodes/20851/downton-abbey-1x01-episode-1",
          name: "Episode 1",
          season: 1,
          number: 1,
          airdate: "2010-09-26",
          airtime: "21:00",
          airstamp: "2010-09-26T20:00:00+00:00",
          runtime: 60,
          image: {
            medium: "http://static.tvmaze.com/uploads/images/medium_landscape/14/36969.jpg",
            original: "http://static.tvmaze.com/uploads/images/original_untouched/14/36969.jpg"
          },
          summary: "<p>Costume drama set in an Edwardian country house in 1912, written by Oscar-winning writer Julian Fellowes.</p>"
        },
        {
          id: "20853",
          url: "http://www.tvmaze.com/episodes/20853/downton-abbey-1x03-episode-3",
          name: "Episode 3",
          season: 1,
          number: 3,
          airdate: "2010-10-10",
          airtime: "21:00",
          airstamp: "2010-10-10T20:00:00+00:00",
          runtime: 60,
          image: {
            medium: "http://static.tvmaze.com/uploads/images/medium_landscape/14/36971.jpg",
            original: "http://static.tvmaze.com/uploads/images/original_untouched/14/36971.jpg"
          },
          summary: "<p>Cora's attempts at matchmaking are dashed when Mary is smitten not by intended target Evelyn Napier, but by his handsome friend, Turkish attache Kemal Pamuk.</p>",
        }
      ]
    }
   ]
};
