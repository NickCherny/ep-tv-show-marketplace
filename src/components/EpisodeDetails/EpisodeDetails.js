import React from 'react';
import {useSelector} from 'react-redux';
import { useParams } from 'react-router-dom';

import { getCurrentEpisode } from '../../store/reducers/shows';

const withEpisodeDetails = ViewComponent => props => {
  const { showId, episodeId } = useParams();
  const episode = useSelector(getCurrentEpisode({ showId, episodeId }));

  return (
    <ViewComponent {...episode} {...props} />
  )
};

export default withEpisodeDetails;
