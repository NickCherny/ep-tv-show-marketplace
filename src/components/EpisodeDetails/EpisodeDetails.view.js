import React from 'react';

import styles from './styles.module.css';

const EpisodeDetails = ({ name, number, summary, image }) => {
  return (
    <div className={styles.EpisodeDetailsContainer}>
      <header className={styles.EpisodeDetailsHeader}>
        <h2 className={styles.EpisodeDetailsTitle}>
          {name}
        </h2>
        <h3 className={styles.EpisodeDetailsArticle}>Episode - {number}</h3>
      </header>
      <article>
        <div className={styles.EpisodeDetailsDescriptionContent} dangerouslySetInnerHTML={{ __html: summary }} />
      </article>
      {Boolean(image) && (
        <div>
          <img src={image.medium} alt={name} />
        </div>
      )}
    </div>
  )
};

export default EpisodeDetails
