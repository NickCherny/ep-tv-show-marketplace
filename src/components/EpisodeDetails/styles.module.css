.EpisodeDetailsContainer {
    padding: 32px 16px;
    display: flex;
    flex-direction: column;
}

.EpisodeDetailsTitle {
    font-size: 24px;
    font-weight: 800;
    color: #282c34;
    margin-bottom: 24px;
}

.EpisodeDetailsHeader {
    margin-bottom: 32px;
}

.EpisodeDetailsArticle {
    display: block;
    font-size: 16px;
    color: #282c34;
}

.EpisodeDetailsDescriptionContent {
    font-size: 16px;
    color: #282c34;
    line-height: 1;
}
