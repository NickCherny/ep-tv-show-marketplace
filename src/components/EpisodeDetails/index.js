import withEpisodeDetails from "./EpisodeDetails";
import EpisodeDetails from "./EpisodeDetails.view";

export default withEpisodeDetails(EpisodeDetails);
