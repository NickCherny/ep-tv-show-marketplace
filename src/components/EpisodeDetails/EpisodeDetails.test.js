import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { __mock_data__ } from '../../setupTests';
import EpisodeDetails from "./index";
import EpisodeDetailsView from './EpisodeDetails.view';

const mockEpisode = __mock_data__.shows[0].episodes[0];

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    showId: '250',
    episodeId: '100'
  }),
  Link: () => <div />
}));

jest.mock('react-redux', () => ({ useSelector: () => mockEpisode }));

describe('<EpisodeDetails />', () => {
  test('props', () => {
    const component = mount(<EpisodeDetails/>);
    const componentView = component.find(EpisodeDetailsView);
    const viewProps = componentView.props();
    expect(viewProps.id).toBe(mockEpisode.id);
  });

  test('snapshot', () => {
    const tree = renderer.create(<EpisodeDetails/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
