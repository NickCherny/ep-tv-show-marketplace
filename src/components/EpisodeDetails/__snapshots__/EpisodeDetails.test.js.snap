// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EpisodeDetails /> snapshot 1`] = `
<div
  className="EpisodeDetailsContainer"
>
  <header
    className="EpisodeDetailsHeader"
  >
    <h2
      className="EpisodeDetailsTitle"
    >
      Cars, Buses and Lawnmowers
    </h2>
    <h3
      className="EpisodeDetailsArticle"
    >
      Episode - 
      1
    </h3>
  </header>
  <article>
    <div
      className="EpisodeDetailsDescriptionContent"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Kirby learns about a character design contest featuring the </p>",
        }
      }
    />
  </article>
</div>
`;
